lib/charms/opensearch/v0/opensearch_backups.py

```diff
@@ -481,38 +490,45 @@ def _on_s3_credentials_changed(self, event: EventBase) -> None:
         if self.s3_client.get_s3_connection_info().get("tls-ca-chain"):
             raise NotImplementedError
 
-        self.charm.status.set(WaitingStatus("Setting up backup service"))
+        self.charm.status.set(MaintenanceStatus(BackupSetupStart))
 
         try:
             plugin = self.charm.plugin_manager.get_plugin(OpenSearchBackupPlugin)
             if self.charm.plugin_manager.status(plugin) == PluginState.ENABLED:
                 self.charm.plugin_manager.apply_config(plugin.disable())
             self.charm.plugin_manager.apply_config(plugin.config())
+        except OpenSearchNotFullyReadyError:
+            logger.warning("s3-changed: cluster not ready yet")
+            event.defer()
+            return
+        except OpenSearchPluginError as e:
+            # This is only possible if we could not apply keys to the store.
+            # That is generally transient.
+            logger.error(e)
+            event.defer()
+            return
         except OpenSearchError as e:
```

Comment:
It seems like these 3 `except`s can be merged in one `except` (they log / defer / return)? and set the status if it's of `PluginError` type?